src/pyhf/writexml.py

```diff
@@ -32,12 +34,11 @@ def _make_hist_name(channel, sample, modifier='', prefix='hist', suffix=''):
 
 
 def _export_root_histogram(histname, data):
-    h = TH1.from_numpy((np.asarray(data), np.arange(len(data) + 1)))
-    h._fName = histname
-    # NB: uproot crashes for some reason, figure out why later
-    # if histname in _ROOT_DATA_FILE:
-    #    raise KeyError(f'Duplicate key {histname} being written.')
-    _ROOT_DATA_FILE[histname] = h
+    hist = TH1.from_numpy((np.asarray(data), np.arange(len(data) + 1)))
+    hist._fName = histname
+    if histname in _ROOT_DATA_FILE:
+        raise KeyError(f"Duplicate key {histname} being written.")
```

Comment:
We're [losing coverage on these two lines as they aren't being tested at the moment](https://codecov.io/gh/scikit-hep/pyhf/pull/1002/diff).

@kratsg This was added as commented out lines way back in PR #426. Do we need this? Or is this a vestige of old problems since resolved?

Reply:
We never had it. I think fundamentally, uproot3 did crash with this. I don't know what uproot4's behavior is meant to be here with duplicated names (@jpivarski ) especially since I don't think the writing functionality exists yet.

For Jim, what is the behavior of uprootX when a histogram stored under key `Foo` is written to a file with that key already existing?

Reply:
The intended behavior for UprootX is that a second object would be written to the ROOT file with a higher cycle number. I think I remember Uproot 3 doing that incorrectly, though.

(The "cycle number" is the number after the semicolon in a list of key names. It disambiguates objects with the same name.)

Reply:
Thanks! I've gone ahead and added a check for this. We generally (for `writexml`) do not want new cycle numbers here, so we need to check if the histogram name exists before adding it to the file as we're essentially building this ROOT file from scratch.